protocol/communities/community_bloom_filter_test.go

```diff
@@ -0,0 +1,68 @@
+package communities
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+
+	"github.com/status-im/status-go/eth-node/crypto"
+	"github.com/status-im/status-go/protocol/common"
+	"github.com/status-im/status-go/protocol/protobuf"
+)
+
+func TestCommunityBloomFilter(t *testing.T) {
+	suite.Run(t, new(CommunityBloomFilterSuite))
+}
+
+type CommunityBloomFilterSuite struct {
+	suite.Suite
+}
+
+func (s *CommunityBloomFilterSuite) TestBasic() {
+	ownerIdentity, err := crypto.GenerateKey()
+	s.Require().NoError(err)
+
+	memberIdentity, err := crypto.GenerateKey()
+	s.Require().NoError(err)
+
+	nonMemberIdentity, err := crypto.GenerateKey()
+	s.Require().NoError(err)
+
+	communityID := "cid"
+	encryptedChannelID := "enc"
+	nonEncryptedChannelID := "non-enc"
+
+	description := &protobuf.CommunityDescription{
+		ID:    communityID,
+		Clock: 1,
+		Chats: map[string]*protobuf.CommunityChat{
+			encryptedChannelID: {
+				Members: map[string]*protobuf.CommunityMember{
+					common.PubkeyToHex(&memberIdentity.PublicKey): {},
+				},
+			},
+			nonEncryptedChannelID: {
+				Members: map[string]*protobuf.CommunityMember{
+					common.PubkeyToHex(&memberIdentity.PublicKey): {},
+				},
+			},
+		},
+		TokenPermissions: map[string]*protobuf.CommunityTokenPermission{
+			"permissionID": {
+				Id:            "permissionID",
+				Type:          protobuf.CommunityTokenPermission_CAN_VIEW_CHANNEL,
+				TokenCriteria: []*protobuf.TokenCriteria{{}},
+				ChatIds:       []string{ChatID(communityID, encryptedChannelID)},
+			},
+		},
+	}
+
+	err = generateBloomFiltersForChannels(description, ownerIdentity)
+	s.Require().NoError(err)
+	s.Require().NotNil(description.Chats[encryptedChannelID].MembersList)
+	s.Require().Nil(description.Chats[nonEncryptedChannelID].MembersList)
+
+	filter := description.Chats[encryptedChannelID].MembersList
+	s.Require().True(testBloomFilter(filter, memberIdentity, &ownerIdentity.PublicKey, encryptedChannelID, description.Clock))
+	s.Require().False(testBloomFilter(filter, nonMemberIdentity, &ownerIdentity.PublicKey, encryptedChannelID, description.Clock))
```

Comment:
Do I get it right that you're explicitly adding a test with 0.001% flakiness? 😄

Reply:
Yes sir :sweat_smile: It is 0.1%.

Reply:
pardon, I mean 0.1%, just confused the numbers 😄

Reply:
Actually, it is much less than that. There are only two items in the filter of n == 128. 0.1% would apply if there were 128 items added.

Reply:
Also, the re-runs on Jenkins!

Reply:
If you run it 100,000 times and it fails even once, I owe you a beer :beer:

Reply:
> If you run it 100,000 times and it fails even once, I owe you a beer 🍺

Let's agree for a beer on each fail 😉

Reply:
```cmd
$ go test -timeout 6000s -tags gowaku_skip_migrations,gowaku_no_rln -count=1000000 -run ^TestCommunityBloomFilter$ github.com/status-im/status-go/protocol/communities
ok      github.com/status-im/status-go/protocol/communities     1532.209s
```

Reply:
![Lucky Bastard](https://media1.tenor.com/m/-VwFkshQMF4AAAAd/friends-ross-geller.gif)